components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java

```diff
@@ -75,8 +82,18 @@ public void validate() {
             throw new IllegalStateException("Sheet name cannot contain dots (.)");
         }
 
-        if (file.exists() && !overwriteFileIfExists) {
-            throw new IllegalStateException("The file already exists and the columns selected do not match. Please configure the job to overwrite the existing file.");
+        if (file.exists()) {
+            Datastore datastore = new ExcelDatastore(file.getName(), new FileResource(file), file.getAbsolutePath());
+            final Schema[] schemas = datastore.openConnection().getDataContext().getSchemas();
```

Comment:
the openConnection() call returns a DatastoreConnection. This needs to be closed in a finally block once you're finished with the operation :-)